From 2a4394ad8de62897121e9ce4b3794bc91d884a17 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Tue, 29 Nov 2022 15:57:15 +0100
Subject: [PATCH] Add ImageService

Still used in LuneOS via legacy luna-sysmgr, so add it back.

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

Upstream-Status: Pending
---
 Src/Main.cpp                                              | 8 ++++++++
 files/sysbus/com.webos.service.systemservice.api.json     | 5 +++++
 files/sysbus/com.webos.service.systemservice.service.in   | 2 +-
 .../com.webos.service.systemservice.service.role.json.in  | 8 +++++++-
 4 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/Src/Main.cpp b/Src/Main.cpp
index 1dc2fac..45fe5aa 100644
--- a/Src/Main.cpp
+++ b/Src/Main.cpp
@@ -27,6 +27,7 @@
 
 #include "SystemRestore.h"
 #include "Mainloop.h"
+#include "ImageServices.h"
 #include "TimeZoneService.h"
 #include "OsInfoService.h"
 #include "DeviceInfoService.h"
@@ -259,6 +260,13 @@ int main(int argc, char ** argv)
 	ClockHandler clockHandler;
 	setupClockHandler(clockHandler, serviceHandle);
 
+	//init the image service
+	ImageServices *imgSvc = ImageServices::instance();
+	imgSvc->init(g_mainloop.get());
+	if (!imgSvc) {
+		qCritical() << "Image service failed init!";
+	}
+
 	//init the timezone service;
 	TimeZoneService* time_zone_srv = TimeZoneService::instance();
 	time_zone_srv->setServiceHandle(serviceHandle);
diff --git a/files/sysbus/com.webos.service.systemservice.api.json b/files/sysbus/com.webos.service.systemservice.api.json
index 54a3c18..e1522c7 100644
--- a/files/sysbus/com.webos.service.systemservice.api.json
+++ b/files/sysbus/com.webos.service.systemservice.api.json
@@ -51,5 +51,10 @@
         "com.webos.service.systemservice/time/getCurrentTimeZoneByLocale",
         "com.webos.service.systemservice/timezone/getTimeZoneFromEasData",
         "com.webos.service.systemservice/timezone/getTimeZoneRules"
+  ],
+  "image.management": [
+        "com.webos.service.image/convert",
+        "com.webos.service.image/imageInfo",
+        "com.webos.service.image/resize"
   ]
 }
diff --git a/files/sysbus/com.webos.service.systemservice.service.in b/files/sysbus/com.webos.service.systemservice.service.in
index 575200f..17f03bd 100644
--- a/files/sysbus/com.webos.service.systemservice.service.in
+++ b/files/sysbus/com.webos.service.systemservice.service.in
@@ -1,4 +1,4 @@
 [D-BUS Service]
-Name=com.webos.service.systemservice
+Name=com.webos.service.systemservice;com.webos.service.image
 Exec=@WEBOS_INSTALL_SBINDIR@/LunaSysService
 Type=static
diff --git a/files/sysbus/com.webos.service.systemservice.service.role.json.in b/files/sysbus/com.webos.service.systemservice.service.role.json.in
index 99b42d8..e309770 100644
--- a/files/sysbus/com.webos.service.systemservice.service.role.json.in
+++ b/files/sysbus/com.webos.service.systemservice.service.role.json.in
@@ -2,7 +2,7 @@
 	"exeName":"@WEBOS_INSTALL_SBINDIR@/LunaSysService",
         "trustLevel" : "oem",
 	"type": "regular",
-	"allowedNames": ["com.webos.service.systemservice"],
+	"allowedNames": ["com.webos.service.systemservice", "com.webos.service.image"],
 	"permissions": [
 		{
 			"service":"com.webos.service.systemservice",
@@ -13,6 +13,12 @@
 				"com.webos.service.alarm",
 				"com.webos.service.applicationManager"
 			]
+		},
+		{
+			"service":"com.webos.service.image",
+			"outbound":[
+				"com.palm.*"
+			]
 		}
 	]
 }
